phoenix-core/src/it/java/org/apache/phoenix/rpc/UpdateCacheIT.java

```diff
@@ -269,4 +317,49 @@ public void testInvalidConnUpdateCacheFrequencyShouldThrow() throws Exception {
             }
         }
     }
+
+    /**
+     * Helper Runnable impl class that continuously keeps checking
+     * if SYSTEM.MUTEX contains any record until either interrupted or
+     * provided connection is closed
+     */
+    private static class SystemMutexCaller implements Runnable {
+
+        private final AtomicBoolean isSysMutexEmpty;
+        private final Connection conn;
+
+        public SystemMutexCaller(final AtomicBoolean isSysMutexEmpty,
+                final Connection conn) {
+            this.isSysMutexEmpty = isSysMutexEmpty;
+            this.conn = conn;
+        }
+
+        @Override
+        public void run() {
+            try {
+                while (!Thread.interrupted() && !conn.isClosed()) {
+                    try {
+                        ResultSet resultSet = conn.createStatement().executeQuery(
+                            "SELECT * FROM " + PhoenixDatabaseMetaData.SYSTEM_MUTEX_NAME);
```

Comment:
In case we _do_ end up going this route then, just to be safe, can we query the exact expected row in SYSTEM.MUTEX instead of this query? If this gets run in parallel with any of the upgrade tests (where we acquire an upgrade mutex), your test will fail.